app/src/main/java/com/bashkevich/androidfundamentals/movieslist/view/MoviesListAdapter.kt

```diff
@@ -43,7 +44,11 @@ class MoviesListAdapter(private val onMovieClickListener: OnMovieClickListener)
         holder.duration.text = holder.context.getString(R.string.duration, movie.runtime)
 
         holder.itemView.setOnClickListener {
-            onMovieClickListener.onMovieClick(movie.id)
+            if (movies.map { it.id }.contains(movie.id)) {
+                onMovieClickListener.onMovieClick(movie.id)
+            } else {
+                Toast.makeText(holder.context, "No such movie!", Toast.LENGTH_SHORT).show()
```

Comment:
Wait how could it be that movie not found?

Reply:
Maybe this movie isn't in DB by magic . I made it just to handle possible error